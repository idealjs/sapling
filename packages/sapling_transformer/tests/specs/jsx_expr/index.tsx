let some = () => {
  return 1;
};

function Expr() {
  const value = some();

  return <div>{value}</div>;
}

const Expr2 = () => {
  return <div>{some()}</div>;
};

const Expr3 = () => {
  const some = () => {
    return 2;
  };
  return <div>{some()}</div>;
};

const Expr4 = () => {
  const some = () => {
    return 3;
  };
  return (
    <div>
      <div>{some()}</div>
    </div>
  );
};
