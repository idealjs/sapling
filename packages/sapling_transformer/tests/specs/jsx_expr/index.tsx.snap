---
source: packages/sapling_transformer/tests/test_utils.rs
expression: index.tsx
---
# Input
```tsx
let some = () => {
  return 1;
};

function Expr() {
  const value = some();

  return <div>{value}</div>;
}

```

# Transformations
```tsx
let some = () => {
  return 1;
};

function Expr() {
  const value = some();
  return (() => {
    let _el$ = _$createElement("div");
    _$insert(_el$, value);
    return _el$;
  })();
}

```
